hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Alert.java

```diff
@@ -89,6 +91,22 @@
     @JsonInclude
     private String resolvedNotes;
 
+    /*
+     * This is the trigger defined when the alert was fired.
+     * A trigger definition can change during time, but an alert should be attached with a specific instance.
+     */
+    @JsonInclude(Include.NON_EMPTY)
+    @Thin
+    private Trigger trigger;
+
+    /*
+     * This is the dampening attached to a trigger when the alert was fired.
+     * As a trigger, the dampening can change during time, but an alert should be attached with a specific instance.
+     */
+    @JsonInclude(Include.NON_EMPTY)
+    @Thin
+    private Dampening dampening;
+
     @JsonInclude(Include.NON_EMPTY)
```

Comment:
Well, the point is that when an alert is created we have a reference of the triggerId but the trigger can be changed. So, in other words, we can have alerts with no info about trigger details or dampening data that originated (we have the conditions, but we miss this info).
Also, another point to maintain that, is that an action should now the alert and the context that generated in one single call, if after receive an alert, the plugin should call to DefinitionsService to get the Trigger and the Dampening in two calls, the overhead can be even higher.
I agree that the object is getting more fat than expected, perhaps after some feedback from the plugins usecase we can just add the significant info that an action should know to have a good knowledge about context when an alert was generated.